Python/pylifecycle.c

```diff
@@ -1921,12 +1921,15 @@ Py_FinalizeEx(void)
     }
 
     if (dump_refs) {
-        _Py_PrintReferences(stderr);
+        _Py_PrintReferences(tstate->interp, stderr);
     }
 
     if (dump_refs_fp != NULL) {
-        _Py_PrintReferences(dump_refs_fp);
+        _Py_PrintReferences(tstate->interp, dump_refs_fp);
     }
+
+    PyObject refchain = { 0 };
+    _Py_StealRefchain(tstate->interp, &refchain);
```

Comment:
Why is this necessary?

Reply:
The interpreter will be deallocated before we later call `_Py_PrintReferenceAddresses()` to walk the linked list.

That said, I think there's a related problem that I still need to address.  Namely, every remaining object in the linked list will be (effectively) freed when the interpreter's allocator is finalized, so the `_ob_next` and `_ob_prev` accesses might segfault.  It likely isn't a big problem *during* runtime finalization (except perhaps for embedders) but I'll fix that regardless.

Reply:
Hmm, I'm pretty sure that's not actually a problem.  We only call `_Py_PrintReferenceAddresses()` in `Py_FinalizeEx()` and only for the main interpreter, which is statically allocated.  Objects in the linked list might still cause a segfault but only if they would have before this change, so I'm not going to worry about that here.  I will leave a note in the code though.